pyvista/trame/jupyter.py

```diff
@@ -0,0 +1,275 @@
+        viewer.ui(
+            mode=mode,
+            default_server_rendering=default_server_rendering,
+            collapse_menu=collapse_menu,
+            **kwargs,
+        )
+
+    return viewer
+
+
+def show_trame(
+    plotter,
+    mode=None,
+    name=None,
+    server_proxy_enabled=None,
+    server_proxy_prefix=None,
+    collapse_menu=False,
+    default_server_rendering=True,
+    return_viewer=False,
+    **kwargs,
+):
+    """Run and display the trame application in jupyter's event loop.
+
+    plotter : pyvista.BasePlotter
+        The PyVista plotter to show.
+
+    mode : str, optional
+        The UI view mode. This can be set on the global theme. Options are:
+            * ``'trame'``: Uses a view that can switch between client and server
+              rendering modes.
+            * ``'server'``: Uses a view that is purely server rendering.
+            * ``'client'``: Uses a view that is purely client rendering (generally
+              safe without a virtual frame buffer)
+
+    name : str, optional
+        The name of the trame server on which the UI is defined
+
+    server_proxy_enabled : bool, default: False
+        Build a relative URL for use with ``jupyter-server-proxy``.
+
+    server_proxy_prefix : str, optional
+        URL prefix when using ``server_proxy_enabled``. This can be set globally in
+        the theme. To ignore, pass ``False``. For use with
+        ``jupyter-server-proxy``, often set to ``proxy/``.
+
+    collapse_menu : bool, default: False
+        Collapse the UI menu (camera controls, etc.) on start.
+
+    default_server_rendering : bool, default: True
+        Whether to use server-side or client-side rendering on-start when
+        using the ``'trame'`` mode.
+
+    return_viewer : bool, optional
+        Return the ipywidget.
+
+    **kwargs : dict, optional
+        Mostly ignored, though ``protocol`` and ``host`` can be use to
+        override the iframe src url and ``hieght`` and ``width`` can be
+        used to overrid the iframe style.
+
+    """
+    if plotter.render_window is None:
+        raise RuntimeError(CLOSED_PLOTTER_ERROR)
+
+    if name is None:
+        server = get_server(name=pyvista.global_theme.trame.jupyter_server_name)
+    else:
+        server = get_server(name=name)
+    if name is None and not server.running:
+        elegantly_launch(server)
+        if not server.running:
+            raise TrameJupyterServerDownError()
+    elif not server.running:
+        raise TrameServerDownError(name)
+
+    # Initialize app
+    viewer = initialize(
+        server,
+        plotter,
+        mode=mode,
+        default_server_rendering=default_server_rendering,
+        collapse_menu=collapse_menu,
+    )
+
+    # Show as cell result
+    src = build_url(
+        server,
+        ui=plotter._id_name,
+        server_proxy_enabled=server_proxy_enabled,
+        server_proxy_prefix=server_proxy_prefix,
+        **kwargs,
+    )
```

Comment:
... Add this
```suggestion
    )
    for ki, key in enumerate(('width', 'height')):
        if key not in kwargs:
            kwargs[key] = f'{plotter.window_size[ki]}px'
```
Or something similar somewhere. FWIW this code fixes the window sizing for MNE-Python

![Screenshot from 2023-01-26 12-33-12](https://user-images.githubusercontent.com/2365790/214907818-c68335d6-ab72-4964-9336-4a3d98cbd5b5.png)

Reply:
Ahh, nice idea!

Reply:
Ehh... I really want the default to be `100%` rather than matching the plotter size. With these changes it makes the output not so elegant by default:

<img width="1071" alt="Screen Shot 2023-01-26 at 11 45 02 AM" src="https://user-images.githubusercontent.com/22067021/214922802-c026870c-0acf-4137-8913-d0b985ea1cbd.png">

Where as with `width: 100%; height: 600px;` its much nicer no matter the window size:

<img width="1071" alt="Screen Shot 2023-01-26 at 11 47 51 AM" src="https://user-images.githubusercontent.com/22067021/214923262-826c4e85-1b51-4c1a-9a4c-e2db00eac8ba.png">

Reply:
I reverted this suggestion

Reply:
But didn't it look that way with `ipyvtklink` already, suggesting that this maintains previous behavior?

Okay with me in principle for 100% -- 50% didn't seem to work all that well -- *assuming there is some nice way in the public API to easily specify a fixed size*. What is it with the current code? It seems like the current version completely ignores `Plotter(..., window_size=...)` which seems unexpected/not backward compatible...

One way out of this might be to have `Plotter(..., window_size='auto')` as a backward compat default which means "`(600px, 600px)` for all backends except trame, which will use `('100%', '600px')`" or something.

Reply:
@larsoner, are you okay with me merging as is and following up in a separate PR (before the release) to expose a better mechanism for controlling the widget size?

Reply:
Okay

Reply:
Tracking here: https://github.com/pyvista/pyvista/issues/3879

Will try to address ASAP